### PR TITLE
pythonPackages.pelican: Fix hash, disable broken tests

### PR DESCRIPTION
**Pull request by johbo**
Faced some trouble building pelican, these tweaks did the trick for me.

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -14114,7 +14114,7 @@ in modules // {
       owner = "getpelican";
       repo = "pelican";
       rev = version;
-      sha256 = "1k572anw39rws67mvxl2w6y93y8w8q5smnwc0dd2gnnr16cc2vsh";
+      sha256 = "1sbnv8dvnlss3hx8bckq40mwk70y46ix422q7lhl8p3rqxsw7zma";
```
Comment (johbo) [inline pkgs/top-level/python-packages.nix]: Hmm, that's now weird for me.

```
output path ‘/nix/store/wy3ahf0bhjp70qg5s15wr8sv603alygq-pelican-3.6.3-src’ has r:sha256 hash ‘1k572anw39rws67mvxl2w6y93y8w8q5smnwc0dd2gnnr16cc2vsh’ when ‘1sbnv8dvnlss3hx8bckq40mwk70y46ix422q7lhl8p3rqxsw7zma’ was expected
[...]
cannot build derivation ‘/nix/store/vgc7vmig39ndjlxclmpn9jfb6fq1c7pr-python2.7-pelican-3.6.3.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/vgc7vmig39ndjlxclmpn9jfb6fq1c7pr-python2.7-pelican-3.6.3.drv’ failed
The invocation of "nix-build -A python35Packages.pelican -A python27Packages.pelican /home/travis/.nox/nixpkgs" failed
```

**Comment by johbo** [comment]
Trying to reproduce this locally, I run into the following result:
```
$ nix-prefetch-zip https://github.com/getpelican/pelican/archive/3.6.3.tar.gz 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   125    0   125    0     0     77      0 --:--:--  0:00:01 --:--:--    91
100  756k  100  756k    0     0  33974      0  0:00:22  0:00:22 --:--:-- 39380
hash is 1sbnv8dvnlss3hx8bckq40mwk70y46ix422q7lhl8p3rqxsw7zma
path is /nix/store/qsivm49gjg919748d15iy6gf63nkyz45-3.6.3.tar.gz
1sbnv8dvnlss3hx8bckq40mwk70y46ix422q7lhl8p3rqxsw7zma
```

Wondering if travis might see a cached old version? Or if there is a way go get a different file from the same location due to another reason?

**Comment by lethalman** [comment]
I can build pelican fine from current master. Is this problem still current?

**Comment by joachifm** [comment]
The hash does not appear to need fixing:
```
$ nix-prefetch-zip --hash-type sha256 https://github.com/getpelican/pelican/archive/3.6.3.tar.gz                                                                             
hash is 1k572anw39rws67mvxl2w6y93y8w8q5smnwc0dd2gnnr16cc2vsh
path is /nix/store/vh1gf1rbzp39maij12k9247p3m7qsqjb-3.6.3.tar.gz
```

I have built pelican for both python27 and python35. I think this can be closed.

**Comment by johbo** [comment]
Hmm, that's weird. I tried to reproduce this on Darwin and on NixOS and I got different results. Running on NixOS, I can confirm that the hash is correct. Now I wonder why the hashes differ.

Here is what I got in a NixOS VM:
```
$ nix-prefetch-zip --hash-type sha256 https://github.com/getpelican/pelican/archive/3.6.3.tar.gz
hash is 1k572anw39rws67mvxl2w6y93y8w8q5smnwc0dd2gnnr16cc2vsh
path is /nix/store/vh1gf1rbzp39maij12k9247p3m7qsqjb-3.6.3.tar.gz
1k572anw39rws67mvxl2w6y93y8w8q5smnwc0dd2gnnr16cc2vsh
```

Running the same command on Darwin, I get this result:
```
$ nix-prefetch-zip --hash-type sha256 https://github.com/getpelican/pelican/archive/3.6.3.tar.gz
hash is 1sbnv8dvnlss3hx8bckq40mwk70y46ix422q7lhl8p3rqxsw7zma
path is /nix/store/qsivm49gjg919748d15iy6gf63nkyz45-3.6.3.tar.gz
1sbnv8dvnlss3hx8bckq40mwk70y46ix422q7lhl8p3rqxsw7zma
```

I will try to find out if that is maybe just happening on my specific box or a general thing and report back here later today.

**Comment by johbo** [comment]
I've tried a different URL and saw consistent hashes:

```
# Command
 nix-prefetch-zip --hash-type sha256 https://github.com/pycurl/pycurl/archive/REL_7_43_0.tar.gz
 
# Darwin
hash is 0dq7mp4vq1mvw7c9ccpjnx2nsqqj0b2ai9zj7hw6jmk5iswdssjr

# NixOS
hash is 0dq7mp4vq1mvw7c9ccpjnx2nsqqj0b2ai9zj7hw6jmk5iswdssjr
```

**Comment by joachifm** [comment]
If you have both tarballs handy, check sizes, unpack & diff to see what's up.

**Comment by johbo** [comment]
I got a bit closer by running `nix-hash` to track down the difference, it seems to originate from a file name which has non-ascii characters in its name. 

The path is `pelican/tests/output/custom_locale/posts/2010/décembre`.

I did generate hashes with a for loop and observed that the result slightly differs when I am in `pelican/tests/output/custom_locale/posts/2010`.

This is my little test loop:
```
for x in *; do printf "%30s %s\n" $x $(nix-hash --type sha256 --base32 $x); done | sort -bf > ~/wo/nix-misc/pr-12983-hashes-pelican/hashes-06-darwin
```
I'll check if my tools are up to date, maybe this got already addressed. If that does not help, then I'll try to create a super small case which reproduces this problem. This does not seem to be python related and changing the hash seems to be the wrong reaction to this problem.

**Comment by johbo** [comment]
Checked with nix version 1.11.2, still there.

Here is how I reproduced it:

```
mkdir reproduce
touch reproduce/décembre
nix-hash reproduce
```

Darwin gives a different hash that NixOS.

Should I close this issue and open up a new one on the Nix repo? After these experiments I assume that the issue sits there.

**Comment by lethalman** [comment]
@johbo seems so... it appears to be a problem with filesystem encoding, or locales.

**Comment by johbo** [comment]
Ok, doing that. Thanks for the support to everyone! 👍

**Pull request closed without merging**